Essentials/src/main/java/com/earth2me/essentials/IUser.java

```diff
@@ -223,4 +236,74 @@
     void setToggleShout(boolean toggleShout);
 
     boolean isToggleShout();
+
+    /**
+     * Gets information about the most-recently-made, non-expired TPA request in the tpa queue of this {@link IUser}.
+     * <p>
+     * The TPA Queue is Last-In-First-Out queue which stores all the active pending teleport
+     * requests of this {@link IUser}. Timeout calculations are also done during the
+     * iteration process of this method, ensuring that teleport requests made past the timeout
+     * period are removed from queue and therefore not returned here. The maximum size of this
+     * queue is determined by {@link ISettings#getTpaMaxAmount()}.
+     *
+     * @param inform             true if the underlying {@link IUser} should be informed if a request expires during iteration.
+     * @param performExpirations true if this method should not spend time validating time for all items in the queue and just return the first item in the queue.
+     * @param excludeHere        true if /tphere requests should be ignored in fetching the next tpa token.
+     * @return A {@link TpaRequest} corresponding to the next available request or null if no valid request is present.
+     */
+    @Nullable TpaRequest getNextTpaRequest(boolean inform, boolean performExpirations, boolean excludeHere);
+
+    /**
+     * Whether or not this {@link IUser} has any valid TPA requests in queue.
+     *
+     * @param inform      true if the underlying {@link IUser} should be informed if a request expires during iteration.
```

Comment:
```suggestion
     * @param inform      true if the user should be informed if a request expires during iteration.
```

The "underlying" user in question is *this* user, no?